helix-core/src/main/java/org/apache/helix/controller/stages/IntermediateStateCalcStage.java

```diff
@@ -865,7 +865,9 @@ private void computeIntermediateMap(PartitionStateMap intermediateStateMap,
         if (!value.getToState().equals(HelixDefinedState.DROPPED.name())) {
           intermediateStateMap.setState(entry.getKey(), value.getTgtName(), value.getToState());
         } else {
-          intermediateStateMap.getStateMap().get(entry.getKey()).remove(value.getTgtName());
+          if (intermediateStateMap.getStateMap().containsKey(entry.getKey())) {
```

Comment:
why not use else if instead of inner if statement.

Reply:
Good catch. Was just coding on auto pilot. Updated